modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,317 @@
+                    digest_prefix, b64_body_digest.decode('ascii'))
+            elif hdr_key == HEADER_HOST:
+                value = target_host
+                request_headers_dict[HEADER_HOST] = '{0}'.format(target_host)
+            else:
+                value = headers[hdr_key]
+            signed_headers_dict[hdr_key] = value
+
+        return signed_header_dict, request_headers_dict
+
+    def get_message_digest(self, data):
+        """
+        Calculates and returns a cryptographic digest of a specified HTTP request.
+
+        :param data: The string representation of the date to be hashed with a cryptographic hash.
+        :return: A tuple of (digest, prefix).
+            The digest is a hashing object that contains the cryptographic digest of the HTTP request.
+            The prefix is a string that identifies the cryptographc hash. It is used to generate the
+            'Digest' header as specified in RFC 3230.
+        """
+        if self.signing_scheme in [SCHEME_RSA_SHA512, SCHEME_HS2019]:
+            digest = SHA512.new()
+            prefix = 'SHA-512='
+        elif self.signing_scheme in [SCHEME_RSA_SHA256]:
+            digest = SHA256.new()
+            prefix = 'SHA-256='
+        else:
+            raise Exception("Unsupported signing algorithm: {0}".format(self.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the signing_info.
+
+        :param digest: A hashing object that contains the cryptographic digest of the HTTP request.
+        :return: A base-64 string representing the cryptographic signature of the input digest.
+        """
+        self.load_private_key()
+        sig_alg = self.signing_algorithm
+        if isinstance(self.private_key, RSA.RsaKey):
+            if sig_alg is None or sig_alg == ALGORITHM_RSASSA_PSS:
+                # RSASSA-PSS in Section 8.1 of RFC8017.
+                signature = pss.new(self.private_key).sign(digest)
+            elif sig_alg == ALGORITHM_RSASSA_PKCS1v15:
+                # RSASSA-PKCS1-v1_5 in Section 8.2 of RFC8017.
+                signature = PKCS1_v1_5.new(self.private_key).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(sig_alg))
+        elif isinstance(self.private_key, ECC.EccKey):
+            if sig_alg is None:
+                sig_alg = ALGORITHM_ECDSA_MODE_FIPS_186_3
+            if sig_alg in ECDSA_KEY_SIGNING_ALGORITHMS:
+                signature = DSS.new(self.private_key, sig_alg).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(sig_alg))
+        else:
+            raise Exception("Unsupported private key: {0}".format(type(self.private_key)))
+        return b64encode(signature)
+
+    def get_authorization_header(self, signed_headers, signed_msg):
```

Comment:
Can change this method's name to add a `__` prefix so we can tell that it is not a publicly used method?
See http://python.robasworld.com/private-member-name-mangling/

Reply:
done.